### ISSUE TITLE
gulp.useref v3 error, update generator

### ISSUE THREAD

**Issue by NicholasGlazer** [mwaylabs/generator-m-ionic#288]
I had an error with gulp.useref generator. Ubuntu 15.10, npm 3.4.1, node 5.0.0. 
According to https://github.com/jonkemp/gulp-useref/issues/153, we shold change generator to avoid this error in new version. 
I solved it with code below. 

```javascript
gulp.task('build-app', ['clean', 'inject-all'], function () {
  var assets = $.useref({searchPath: '{.tmp,app}'});
  var jsFilter = $.filter('**/*.js', {restore: true});
  var cssFilter = $.filter('**/*.css', {restore: true});

  var stream = gulp.src('app/index.html') // main html file
    .pipe(assets); // all assets (without index.html)

  if (options.minify) {
    stream
      .pipe(jsFilter)
      .pipe($.ngAnnotate({
        add: true,
        sourcemap: true
      }))
      .pipe($.uglify())
      .pipe(jsFilter.restore)
      .pipe(cssFilter)
      .pipe($.csso())
      .pipe(cssFilter.restore);
  }

  stream
    .pipe(gulp.dest(paths.dist));

  return stream;
});
```

**Comment by gruppjo** [comment]
We're gonna update the generator's dependencies soon and thus also gulp-useref. This might save us some time. Thanks!

**Comment by gruppjo** [comment]
fixed with bd715cb7e312166e3537b50562d722ba67b2a586 on #290.

Thank you very much for the heads-up! :)
